x-pack/plugins/canvas/shareable_runtime/components/canvas.tsx

```diff
@@ -85,9 +85,9 @@ export const CanvasComponent = ({
     // any point, or change the interval, we need to make sure the interval is
     // killed on React re-render-- otherwise the pages will start bouncing around
     // as timeouts are accumulated.
-    clearTimeout(timeout);
+    window.clearTimeout(timeout);
 
-    timeout = setTimeout(
+    timeout = window.setTimeout(
```

Comment:
what caused the need for this change?

Reply:
That's because of NodeJS typings update that might have (finally) made `setTimeout` related types stricter. We don't yet clearly separate environments for TS compiler to know that this `setTimeout` is supposed to be provided by the browser and TS assumes that it's a NodeJS one. NodeJS `setTimeout` is a different API with a different return type (`NodeJS.Timeout` instead of `number`) than what browsers provide.